File: mobile/ios/Infy/Shared/Networking/APIEnvelope.swift
import Foundation

struct APIEnvelope<Payload: Decodable>: Decodable {
    let data: Payload
}

struct APIErrorEnvelope: Decodable {
    let error: APIErrorBody
}

struct APIErrorBody: Decodable, Equatable, Sendable {
    let code: String
    let message: String
}

struct EmptyPayload: Decodable, Sendable {}

File: mobile/android/app/src/main/java/com/infy/messenger/core/network/ApiEnvelope.kt
package com.infy.messenger.core.network

import kotlinx.serialization.Serializable

/**
 * Бэкенд оборачивает успешный ответ в { "data": T }.
 * Используется как тип ответа Retrofit для всех эндпоинтов, что-либо возвращающих.
 */
@Serializable
data class ApiEnvelope<T>(
    val data: T,
)

/** Ошибочный ответ бэкенда: { "error": { "code", "message" } }. */
@Serializable
data class ApiErrorEnvelope(
    val error: ApiErrorBody,
)

@Serializable
data class ApiErrorBody(
    val code: String,
    val message: String,
)
